protocol-select/README.md

```diff
@@ -0,0 +1,541 @@
+
+| Lifecycle Stage | Maturity       | Status | Latest Revision |
+|-----------------|----------------|--------|-----------------|
+| 1A              | Working Draft  | Active | r0, 2021-XX-XX  |
+
+Authors: [@marten-seemann], [@mxinden]
+
+Interest Group:
+
+[@marten-seemann]: https://github.com/marten-seemann
+[@mxinden]: https://github.com/mxinden
+
+See the [lifecycle document][lifecycle-spec] for context about maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+## Table of Contents
+
+- [Protocol Select](#protocol-select)
+    - [Table of Contents](#table-of-contents)
+    - [Introduction](#introduction)
+        - [Improvements over _[Multistream Select]_](#improvements-over-_multistream-select_)
+    - [High-Level Overview](#high-level-overview)
+        - [Basic Flow](#basic-flow)
+        - [Secure Channel Selection](#secure-channel-selection)
+        - [TCP Simultaneous Open](#tcp-simultaneous-open)
+            - [Coordinated TCP Simultaneous Open](#coordinated-tcp-simultaneous-open)
+            - [Uncoordinated TCP Simultaneous Open](#uncoordinated-tcp-simultaneous-open)
+        - [Connection Protocol Negotiation](#connection-protocol-negotiation)
+            - [Early data optimization](#early-data-optimization)
+            - [0-RTT](#0-rtt)
+        - [Stream Protocol Negotiation](#stream-protocol-negotiation)
+            - [Initiator](#initiator)
+            - [Listener](#listener)
+    - [Transitioning from [Multistream Select]](#transitioning-from-multistream-select)
+        - [Multiphase Rollout](#multiphase-rollout)
+            - [Phase 1](#phase-1)
+            - [Phase 2](#phase-2)
+            - [Phase 3](#phase-3)
+        - [Additional Rollout Mechanisms](#additional-rollout-mechanisms)
+        - [Heuristics](#heuristics)
+            - [TCP](#tcp)
+            - [QUIC](#quic)
+            - [Protocol Differentiation](#protocol-differentiation)
+    - [Protocol Specification](#protocol-specification)
+        - [Protocol Evolution](#protocol-evolution)
+            - [Non-Breaking Changes](#non-breaking-changes)
+            - [Breaking Changes](#breaking-changes)
+    - [Extensions](#extensions)
+        - [Protocol IDs](#protocol-ids)
+    - [FAQ](#faq)
+
+## Introduction
+
+_Protocol Select_ is a protocol negotiation protocol. It is aimed at negotiating
+libp2p protocols on connections and streams. It replaces the _[Multistream
+Select]_ protocol.
+
+### Improvements over _[Multistream Select]_
+
+- **Downgrade attacks** and **censorship resistance**
+
+  Given that **[Multistream Select]** negotiates a connection's security
+  protocol unencrypted and unauthenticated it is prone to [downgrade attack]s.
+  In addition, a man-in-the-middle can detect that a given connection is used
+  to carry libp2p traffic, allowing attackers to censor such connections.
+
+  **Protocol Select** is combined with a change to the Multiaddr format,
```

Comment:
(slightly confusing because we're not actually changing the multiaddr _format_)

Reply:
This would be fixed with #354.